documents/converters.go

```diff
@@ -329,30 +331,66 @@ func ToClientCollaboratorAccess(ca CollaboratorsAccess) (*documentpb.ReadAccess,
 }
 
 // DeriveResponseHeader derives common response header for model
-func DeriveResponseHeader(model Model) (*documentpb.ResponseHeader, error) {
+func DeriveResponseHeader(tokenRegistry TokenRegistry, model Model) (*documentpb.ResponseHeader, error) {
 	cs, err := model.GetCollaborators()
 	if err != nil {
 		return nil, errors.NewTypedError(ErrCollaborators, err)
 	}
 
-	rcs, wcs := ToClientCollaboratorAccess(cs)
-	// TODO(ved): we need to update log for NewCoreDocumentWithCollaborators and PrepareNewVersion. Please don't do this without further discussion since this affects consensus validations.
-	//author, err := model.Author()
-	//if err != nil {
-	//	return nil, err
-	//}
-	//
-	//time, err := model.Timestamp()
-	//if err != nil {
-	//	return nil, err
-	//}
+	author := ""
+	a, err := model.Author()
+	// we ignore error here because it can happen when a model us first created but its not anchored yet
+	if err == nil {
+		author = a.String()
+	}
+
+	// we ignore error here because it can happen when a model us first created but its not anchored yet
+	time := ""
+	t, err := model.Timestamp()
+	if err == nil {
+		time = t.UTC().String()
+	}
 
+	nfts := model.NFTs()
+	cnfts, err := convertNFTs(tokenRegistry, nfts)
+	if err != nil {
+		// this could be a temporary failure, so we ignore but warn about the error
```

Comment:
what is the impact of this erroring out?

Reply:
its would prevent someone from getting a document through API